pkg/api/avatar/avatar.go

```diff
@@ -104,8 +105,8 @@ func (this *CacheServer) Handler(ctx *macaron.Context) {
 
 	if avatar.notFound {
 		avatar = this.notFound
-	} else {
-		if err := this.cache.Add(hash, avatar, gocache.DefaultExpiration); err != nil {
+	} else if !exists {
```

Comment:
@marefr Could this function be called by several threads at once? In that case, using this variable is pretty pointless.